pkg/sriov-fec/nodeconfig.go

```diff
@@ -0,0 +1,356 @@
+package sriovfec
+
+import (
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	"github.com/openshift-kni/eco-goinfra/pkg/sriov-fec/sriovfectypes"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	sriovfecnodeconfig = "SriovFecNodeConfig"
+)
+
+// NodeConfigBuilder provides struct for the SriovFecNodeConfig object containing connection to
+// the cluster and the SriovFecNodeConfig definitions.
+type NodeConfigBuilder struct {
+	// SriovFecNodeConfig definition. Used to create SriovFecNodeConfig object.
+	Definition *sriovfectypes.SriovFecNodeConfig
+	// Create SriovFecNodeConfig object.
+	Object *sriovfectypes.SriovFecNodeConfig
+	// apiClient opens a connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define SriovFecNodeConfig definitions. errorMsg is processed before SriovFecNodeConfig
+	// object is created.
+	errorMsg string
+}
+
+// AdditionalOptions additional options for sriovfecnodeconfig object.
+type AdditionalOptions func(builder *NodeConfigBuilder) (*NodeConfigBuilder, error)
+
+// NewNodeConfigBuilder creates a new instance of NodeConfigBuilder.
+func NewNodeConfigBuilder(
+	apiClient *clients.Settings,
+	name, nsname string,
+	label map[string]string) *NodeConfigBuilder {
+	glog.V(100).Infof(
+		"Initializing new SriovFecNodeConfig structure with the following params: %s, %s, %v",
+		name, nsname, label)
+
+	builder := NodeConfigBuilder{
+		apiClient: apiClient,
+		Definition: &sriovfectypes.SriovFecNodeConfig{
+			TypeMeta: metaV1.TypeMeta{
+				Kind:       sriovfecnodeconfig,
+				APIVersion: fmt.Sprintf("%s/%s", APIGroup, APIVersion),
+			},
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the SriovFecNodeConfig is empty")
+
+		builder.errorMsg = "SriovFecNodeConfig 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the SriovFecNodeConfig is empty")
+
+		builder.errorMsg = "SriovFecNodeConfig 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Pull retrieves an existing SriovFecNodeConfig.io object from the cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*NodeConfigBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling SriovFecNodeConfig.io object name: %s in namespace: %s", name, nsname)
+
+	builder := NodeConfigBuilder{
+		apiClient: apiClient,
+		Definition: &sriovfectypes.SriovFecNodeConfig{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
```

Comment:
Small update. We've decided to update Pull function.
Can you please refactor the function to the following format:

```
if name == "" {
                return nil, fmt.Error("The name of the SriovFecNodeConfig is empty")
	}

if nsname == "" {
                return nil, fmt.Error("The namespace of the SriovFecNodeConfig is empty")
	}
```

Reply:
Sure, updated.